src/go/k8s/pkg/resources/certmanager/certificate.go

```diff
@@ -107,6 +107,7 @@ func (r *CertificateResource) obj() (k8sclient.Object, error) {
 
 	if r.fqdn != "" {
 		name := "*." + strings.TrimSuffix(r.fqdn, ".")
+		cert.Spec.CommonName = fmt.Sprintf("%s-node", r.pandaCluster.Name)
```

Comment:
Because the `r.pandaCluster.Name` has the limit of 64 characters the additional `-node` will exceed the limit.

Reply:
Should we add e2e test with very long name?
Maybe change one of current test to have very long name.